### PR TITLE
Vite migration

### PR DESCRIPTION
**Pull request by geekuillaume**
This PR migrate from create-react-app to [Vite](https://vitejs.dev/). Using vite improves the build time in dev and prod mode a lot. This originates from my attempt at upgrading to CRA v5 which was a failure because of the decision the CRA team made to not give any option to include basic polyfills for NodeJS modules the deps in this project are using.

### CODE REVIEW

**Comment by jeffeb3** [comment]
I appreciate these PRs. @bobnik is going to look into this one for me. It looks fine, but I am not a javascript expert.

BTW, we aren't very fast, especially me. But we will get to these PRs.

File: src/features/app/store.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ const persistInitKey = 'state'
 const persistSaveKey = 'state'
 
 // never persist state when running within a jest spec
-if (process.env.JEST_WORKER_ID === undefined && persistState) {
+if (import.meta.env.JEST_WORKER_ID === undefined && persistState) {
```
Comment (bobnik) [inline src/features/app/store.js:41]: @geekuillaume Can you tell me why you had to make this change?

**Comment by bobnik** [comment]
@geekuillaume I've looked over this PR, and I'm OK with moving to Vite. It's a step in the right direction. Your current PR breaks some out of the box CRA functionality, like eslint and jest specs. I've locally made (I think) all of the changes needed to get these features back. If it's alright with you, I'll push some fixes to your PR and we'll iterate until we're all happy.

@jeffeb3 When the PR is ready, you'll have to try out the new build to be sure nothing is broken.

File: src/features/app/store.js (hunk 1 above)
Comment (bobnik) [reply src/features/app/store.js:41]: Never mind. I discovered the reason :)

**Comment by geekuillaume** [comment]
> @geekuillaume I've looked over this PR, and I'm OK with moving to Vite. It's a step in the right direction. Your current PR breaks some out of the box CRA functionality, like eslint and jest specs. I've locally made (I think) all of the changes needed to get these features back. If it's alright with you, I'll push some fixes to your PR and we'll iterate until we're all happy.
> 
> @jeffeb3 When the PR is ready, you'll have to try out the new build to be sure nothing is broken.

You can push on this PR! Happy to see this merged :)

**Comment by bobnik** [comment]
@geekuillaume @jeffeb3 I've pushed my changes. Please review. As mentioned previously, most of my changes were made to (1) completely remove CRA, and (2) manually add eslint and jest along with approximate copies of CRA's configuration for each.

Jeff, per Vite's instructions, I moved favicon.svg and CNAME files to the root folder. I wasn't sure how to test the CNAME change.

**Comment by geekuillaume** [comment]
Awesome! Thanks for this! :)

**Comment by jeffeb3** [comment]
> Jeff, per Vite's instructions, I moved favicon.svg and CNAME files to the root folder. I wasn't sure how to test the CNAME change.

That stuff is all on the edge of my ability. I think we will probably have to pay close attention when we make the next release. But I am sure we can figure it out. 

> Oh, one last thing - lint no longer runs inside the node process. You will need to run it manually (since we don't have it integrated into CI, presumably just remember to run it before you push your changes).

I didn't know we had eslint :). How do I run it?

> When the PR is ready, you'll have to try out the new build to be sure nothing is broken.

I have gotten around to trying this. I am running npm in a docker container to avoid changing the versions on my host OS. That all seems to work, although I am sure I could make it a bit smoother. I will put the dockerfiles up later, in a different PR.

**Pull request merged**